packages/shared/src/lib/auxiliary/wallet-connect/handlers/sign_message.handler.ts

```diff
@@ -48,17 +42,7 @@ export function handleSignMessage(
                 onCancel: () => responseCallback({ error: getSdkError('USER_REJECTED') }),
             },
         })
-
-    if (account.index !== getSelectedAccountIndex()) {
-        openPopup({
-            id: PopupId.DappAccountSwitcher,
-            props: {
-                account,
-                onCancel: () => responseCallback({ error: getSdkError('USER_REJECTED') }),
-                onConfirm: openSignMessagePopup,
-            },
-        })
-    } else {
-        openSignMessagePopup()
+    } catch (err) {
+        responseCallback({ error: getSdkError('USER_REJECTED') })
```

Comment:
in here, we're missing the case if `err === "Request rejected by Wallet"`. optimally we'd want to return `getSdkError('UNSUPPORTED_ACCOUNTS')`